packages/indexer-cli/src/actions.ts

```diff
@@ -101,6 +105,58 @@ export async function validateActionInput(
   )
 }
 
+export function validateActionType(input: string): ActionType {
+  const validVariants = Object.keys(ActionType).map(variant =>
+    variant.toLocaleLowerCase(),
+  )
+  if (!validVariants.includes(input.toLocaleLowerCase())) {
+    throw Error(
+      `Invalid 'ActionType' "${input}", must be one of ['${validVariants.join(`', '`)}']`,
+    )
+  }
+  return ActionType[input.toUpperCase() as keyof typeof ActionType]
+}
+
+export function validateActionStatus(input: string): ActionStatus {
+  const validVariants = Object.keys(ActionStatus).map(variant =>
+    variant.toLocaleLowerCase(),
+  )
+  if (!validVariants.includes(input.toLocaleLowerCase())) {
+    throw Error(
+      `Invalid 'ActionStatus' "${input}", must be one of ['${validVariants.join(
+        `', '`,
+      )}']`,
+    )
+  }
+  return ActionStatus[input.toUpperCase() as keyof typeof ActionStatus]
+}
+
+export function buildActionFilter(
+  id: string | undefined,
+  type: string | undefined,
+  status: string | undefined,
+  source: string | undefined,
+  reason: string | undefined,
+): ActionFilter {
+  const filter: ActionFilter = {}
```

Comment:
I think leaving filter blank (for example `indexer actions update force true`) will update all the actions, is that intended or do you think we can add an extra check for `all`?